### PR TITLE
Discord Rich Presence Feature

### PR DESCRIPTION
**Pull request by DidiSkywalker**
Feel free to make suggestions for and follow progress of the Discord Rich Presence feature suggested in #220 here!

### CODE REVIEW

**Comment by MartinNemi03** [comment]
About showing time and profile details, I was thinking of it being in large and small image text elements instead of state and details elements and maybe toggleable between each other with also an option to just show the current version of skyblock addons it's being powered by. Included with this posts is a great looking graphical presentation of this idea.

obrazek obrazek obrazek

Thanks for getting on the idea and making it possible!

**Comment by DidiSkywalker** [comment]
Oh I like that! I was gonna have the SBA Version fixed for small image text, to have it say 'Playing Skyblock' and still have SBA mentioned somewhere, but the time as large image text is cool. The people from Hypixel Packs HQ also agreed to make some icons for the Skyblock Zones we could use there.
Will probably keep the text options, just for customizations sake, but have time and zone always displayed with the image then.

**Comment by asbyth** [comment]
consider using `com.github.jagrosh:DiscordIPC:e29d6d8` rather than 0.4, as they fixed something in that commit which would prevent the connection/pipe from properly closing, but never officially pushed to master for whatever reason
https://github.com/jagrosh/DiscordIPC/commit/e29d6d85c1e079e9d36f01ee89b7eeeebf40439e

**Comment by biscuut** [comment]
ooh okay thanks asbyth

File: src/main/java/codes/biscuit/skyblockaddons/gui/SettingsGui.java [hunk 1]
```diff
@@ -7,13 +7,15 @@
...
 import codes.biscuit.skyblockaddons.utils.Message;
+import codes.biscuit.skyblockaddons.utils.discord.DiscordStatus;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiButton;
 import net.minecraft.client.gui.GuiScreen;
 import net.minecraft.client.gui.ScaledResolution;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.util.ResourceLocation;
 import net.minecraftforge.client.GuiIngameForge;
+import scala.actors.threadpool.Arrays;
```
Comment (SteveKunG) [inline src/main/java/codes/biscuit/skyblockaddons/gui/SettingsGui.java]: Don't use Arrays from scala.

**Comment by biscuut** [comment]
tisk tisk scala always sneaking in

File: src/main/java/codes/biscuit/skyblockaddons/gui/SettingsGui.java (hunk 1 above)
Comment (DidiSkywalker) [reply src/main/java/codes/biscuit/skyblockaddons/gui/SettingsGui.java]: Oops, definitely not intentional, thanks!

File: src/main/java/codes/biscuit/skyblockaddons/gui/buttons/ButtonSelect.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+                && mouseY < yPosition + height;
+    }
+
+    /**
+     * Replaces the last character in the given string with the ellipses character {@code …}
+     *
+     * @param text Text to ellipsize
+     * @return Input text with … at the end
+     */
+    private String ellipsize(String text) {
```
Comment (ichbineinNerd) [inline src/main/java/codes/biscuit/skyblockaddons/gui/buttons/ButtonSelect.java]: this seems a little pointles…

**Comment by ILikePlayingGames** [comment]
@biscuut merged it manually.

**Pull request closed without merging**